### ISSUE TITLE
I/O error when window is resized while read is running

### ISSUE THREAD

**Issue by Crestwave** [oilshell/oil#292]
```
Traceback (most recent call last):
  File "/home/andy/git/oilshell/oil/bin/oil.py", line 766, in _cpython_main_hook
  File "/home/andy/git/oilshell/oil/bin/oil.py", line 748, in main
  File "/home/andy/git/oilshell/oil/bin/oil.py", line 722, in AppBundleMain
  File "/home/andy/git/oilshell/oil/bin/oil.py", line 534, in ShellMain
  File "/home/andy/git/oilshell/oil/core/main_loop.py", line 114, in Interactive
  File "/home/andy/git/oilshell/oil/osh/cmd_exec.py", line 1234, in ExecuteAndCatch
  File "/home/andy/git/oilshell/oil/osh/cmd_exec.py", line 1182, in _Execute
  File "/home/andy/git/oilshell/oil/osh/cmd_exec.py", line 1127, in _Dispatch
  File "/home/andy/git/oilshell/oil/osh/cmd_exec.py", line 585, in RunSimpleCommand
  File "/home/andy/git/oilshell/oil/osh/cmd_exec.py", line 393, in _RunBuiltin
  File "/home/andy/git/oilshell/oil/osh/builtin.py", line 539, in Read
  File "/home/andy/git/oilshell/oil/osh/builtin.py", line 498, in ReadLineFromStdin
OSError: [Errno 4] Interrupted system call
```

**Comment by andychu** [comment]
What script are you running here?

This behavior has changed since the latest release.  If you try on `master`, the behavior will be different, but it might not be correct still.  If you send the script then I can try it.

**Comment by andychu** [comment]
Yeah the new behavior isn't right either... I think this has to do with the fact that Python has some builtin handline for `EINTR`, but then when we register `SIGWINCH`, that logic doesn't come into play?  Not sure yet.

**Comment by Crestwave** [comment]
The script I was originally running was the BF interpreter written in `sh` on `master`; I got `osh I/O error: Interrupted system call` there. This one was done interactively on the latest release.

**Comment by andychu** [comment]
A similar thing happens with `sleep 5 | wc -l` .  Resizing the window causes it to return :-(

Clearly SIGWINCH handling has to completely change ... maybe let GNU readline handle it.

**Comment by andychu** [comment]
Fixed with 0.6.pre22: http://www.oilshell.org/release/0.6.pre22/changelog.html
